lib/split/user.rb

```diff
@@ -14,12 +14,10 @@ def initialize(context, adapter=nil)
 
     def cleanup_old_experiments!
       return if @cleaned_up
-      experiment_keys(user.keys).each do |key|
-        experiment = ExperimentCatalog.find key_without_version(key)
+      user.keys.each do |key|
+        experiment = ExperimentCatalog.find experiment_name(key)
```

Comment:
`experiment_name` is pretty similar to `key_without_version`, except `key_without_version` can't handle the case like `exp805:eligibility`.  I thought about changing the implementation of `key_without_version`. But because this private method is from the original Split fork, I'm not sure if I should change it.

Reply:
I think it's good that we're removing potentially dangling keys, although I think we're potentially making unnecessary Redis calls here for multiple keys of the same experiment ~ perhaps we should create/store a temporary hash so we don't have to call Redis between iterations of the same experiment.